crates/sui-indexer/migrations/2023-10-06-204335_tx_indices/up.sql

```diff
@@ -3,47 +3,64 @@ CREATE TABLE tx_senders (
     tx_sequence_number          BIGINT       NOT NULL,
     -- SuiAddress in bytes.
     sender                      BYTEA        NOT NULL,
+    -- SystemTransaction/ProgrammableTransaction. See types.rs
+    transaction_kind            smallint     NOT NULL,
     PRIMARY KEY(sender, tx_sequence_number, cp_sequence_number)
 );
 CREATE INDEX tx_senders_tx_sequence_number_index ON tx_senders (tx_sequence_number, cp_sequence_number);
+CREATE INDEX tx_senders_transaction_kind_index ON tx_senders (sender, transaction_kind, tx_sequence_number, cp_sequence_number);
 
 CREATE TABLE tx_recipients (
     cp_sequence_number          BIGINT       NOT NULL,
     tx_sequence_number          BIGINT       NOT NULL,
     -- SuiAddress in bytes.
     recipient                   BYTEA        NOT NULL,
+    -- SystemTransaction/ProgrammableTransaction. See types.rs
+    transaction_kind            smallint     NOT NULL,
     PRIMARY KEY(recipient, tx_sequence_number, cp_sequence_number)
 );
 CREATE INDEX tx_recipients_tx_sequence_number_index ON tx_recipients (tx_sequence_number, cp_sequence_number);
+CREATE INDEX tx_recipients_transaction_kind_index ON tx_recipients (recipient, transaction_kind, tx_sequence_number, cp_sequence_number);
 
 CREATE TABLE tx_input_objects (
     cp_sequence_number          BIGINT       NOT NULL,
     tx_sequence_number          BIGINT       NOT NULL,
-    -- Object ID in bytes. 
+    -- Object ID in bytes.
     object_id                   BYTEA        NOT NULL,
+    address                     BYTEA        NOT NULL,
```

Comment:
each tx lookup table will also have `address` and `rel` to facilitate graphql queries filtered on address + the lookup. These are additive changes and should still be backwards compatible.